### PR TITLE
Change bad rows file to a bad rows report

### PR DESCRIPTION
**Pull request by martonpe**

### CODE REVIEW

**Comment by martonpe** [comment]
Tests are failing even though it works as expected if not ran with `rake spec`. I'll investigate.

**Comment by tmichel** [comment]
I took a quick look and it seems that the modified [`Avro::Schema`](https://github.com/sspinc/csv2avro/blob/cbd85938f181379a784f4d28428b4bed910e0e45/lib/avro_schema.rb) does not clear it's `errors` array and basically you print out all errors that ever occurred till that point. 

I have now explanation why it works when you run it without RSpec. How do you run it? The same input as the in the spec?

**Comment by gszjulcsi** [comment]
LGTM code wise, but let's figure out why CircleCI is failing.

**Comment by martonpe** [comment]
This finally works and tests run as expected. @tmichel can you merge if it looks good?

File: csv2avro.gemspec [hunk 1]
```diff
@@ -23,6 +23,7 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency "rspec", "~> 3.2"
   spec.add_development_dependency "pry", "~> 0.10"
   spec.add_development_dependency "bump", "~> 0.5"
+  spec.add_development_dependency "byebug"
```
Comment (rgabo) [inline csv2avro.gemspec]: Do we want to keep this development dependency around?

File: spec/csv2avro/converter_spec.rb [hunk 2]
```diff
@@ -351,15 +351,15 @@
         CSV2Avro::Converter.new(reader, avro_writer, bad_rows_writer, error_writer, { delimiter: "\t" }, schema: schema).convert
       end
 
-      it 'should have the bad data in the original form' do
+      it 'should report the bad rows correctly' do
         expect(bad_rows_writer.string).to eq(
-          "id\ttitle\tdescription\n1\t\tdresses\n4\t\tfemale-shoes\n"
+          "row 2: Missing value at name\nrow 5: Missing value at name\n"
```
Comment (rgabo) [inline spec/csv2avro/converter_spec.rb]: I would omit the word `row` or replace it with `L2:` which is a standard way of saying "line 2". What's the most readable for whoever looks at this file? I also think a file with the syntax:

```
L2: Missing value at name
```
could be used by editors to navigate to the errors.

What does JSLint and co. output?

**Comment by rgabo** [comment]
Great piece of refactoring, LGTM with comments

File: spec/csv2avro/converter_spec.rb (hunk 2 above)
Comment (martonpe) [inline spec/csv2avro/converter_spec.rb]: :+1: I like the L2 idea.

File: csv2avro.gemspec (hunk 1 above)
Comment (martonpe) [inline csv2avro.gemspec]: Well, it is useful and doesn't hurt anyone, does it? :)

Comment (rgabo) [inline csv2avro.gemspec]: I think it's unneccessary :)

Comment (martonpe) [inline csv2avro.gemspec]: Ok, I'll remove :)

**Pull request merged**